src/main/webapp/script.js

```diff
@@ -10,11 +15,49 @@ function passQuery() {
         .then((data) => { 
           // data is a 2D array, where the first row is a header row and all
           // subsequent rows are one piece of data (e.g. for a state or county)
-          callOliviaFunction(data);
+          displayVisualization(data);
         });
       }
       else {
         console.log("There was an error");
       }
     });
-}
\ No newline at end of file
+}
+
+ // displayVisualization takes in a data array representing the 2D array returned by the 
+ // census API. The function creates the necessary visualization with the census data.
+function displayVisualization(censusDataArray) {
+  if (censusDataArray[0][2] == 'state' && censusDataArray[0][3] != 'county') {
+    google.charts.setOnLoadCallback(drawRegionsMap(censusDataArray));
+  } else {
+    // We currently do not have counties implemented
+    document.getElementById('map').innerHTML = "";
+    document.getElementById('more-info').innerHTML = 'We do not have this information yet.'
+  }
+}
+
+// Takes in a 2D array from the census API and draws the appropriate visualization
+function drawRegionsMap(censusDataArray) {
+  const shortDataArray = createDataArray(censusDataArray);
+  var data = google.visualization.arrayToDataTable(shortDataArray);
+  var options = {
+    'region': 'US',
+    'resolution': 'provinces',
+    'colorAxis': {colors: ['white', 'blue']}
+  };
+  var chart = new google.visualization.GeoChart(document.getElementById('map'));
+  chart.draw(data, options);
+  document.getElementById('more-info').innerHTML = 'These populations are divided by 1000';
```

Comment:
Populations(In Thousands) or something similar  might be better.

Reply:
fixed